Add `reverse` support to time scale

kurkle opened the pull request:
Simpler approach for reversing time scale.
<del>What I don't know is why those labels rotate, that needs to be investigated still.</del>

Tests adopted from PR: #5731 

![3279](https://user-images.githubusercontent.com/27971921/50244832-9ab74780-03d9-11e9-9cfe-e55113b24963.png)
(tick rotation fixed in fiddle)

2.7.3: https://jsfiddle.net/rr3ftzge/40/
PR: https://jsfiddle.net/9cr418ty/1/
Updated PR: https://jsfiddle.net/9cr418ty/2/

Closes #5731
Fixes #3279

simonbrunel commented on src/scales/scale.time.js:
```diff
@@ -383,7 +383,7 @@ function computeOffsets(table, ticks, min, max, options) {
 	}
 
-	return {left: left, right: right};
+	return options.ticks.reverse ? {left: right, right: left} : {left: left, right: right};
```
Not directly related to this PR but maybe the good timing to change it: since we use these values for vertical and now reversed scales, it may be easier to understand if we rename those properties for `start` and `end`:

```javascript
return options.ticks.reverse ? {start: start, end: end} : {start: end, end: start};
```

kurkle commented:
Changed start to end as @simonbrunel suggested and noticed and fixed a issue.

![5927](https://user-images.githubusercontent.com/27971921/50276742-59f81680-044b-11e9-8f73-2360e4965de1.gif)

simonbrunel commented:
Does it work correctly with other `bound` and `distribution` option values?

kurkle commented:
> Does it work correctly with other `bound` and `distribution` option values?

It does:
https://codepen.io/kurkle/full/gZgGWr
